modules/mongodb/types.go

```diff
@@ -53,6 +54,11 @@ func (conn *Connection) ReadMsg() ([]byte, error) {
 		return nil, err
 	}
 	msglen := binary.LittleEndian.Uint32(msglen_buf[:])
+	if msglen < 4 {
```

Comment:
This looks good, but it also brings up the opposite extreme; if the spec has a limit (other than 2^32 - 1), that is always a good choice, as long as it's a "reasonable" size; if not, it's a tradeoff between what use you can get from reading huge chunks of data vs the cost in time / bandwidth / memory / storage of doing so, and the likelihood of getting a genuine MongoDB server of that size vs random data (e.g. what are the odds that a "size" greater than 0x20000000 is actually a valid response vs it's the odds that it's actually ASCII text?).

Reply:
Maybe we should just add some arbitrary limit to avoid allocating giant buffers in case something happens to send back a 0xffffffff.